File: calculadora.py

from unittest import result


def calculadora(num1, num2, operacao):
    if(operacao == 1): return num1+num2
    elif(operacao == 2): return num1-num2
    elif(operacao == 3): return num1*num2
    elif(operacao == 4): return num1/num2
    else: return False
num1 = int(input("digite o primeiro número"))
num2 = int(input("digite o segundo número"))
operacao= int(input("digite um número entre 1 e 4, sabendo que: 1=soma;2=subtração;3=multiplicação;4=divisão"))
resultado = calculadora(num1,num2, operacao)
print(resultado)

File: trilha.3_js/atividade08/calculadora.js
try {
    
    const num1 = parseFloat(prompt('Digite um número número:'))
    const op = prompt('informe a operação a ser realizada:')
    const num2 = parseFloat(prompt('Digite outro número:'))
    

    if(op == "soma"){
        let resultado = num1 + num2
        console.log(`Resultado ${resultado}`)
    } else if(op === "subtração"){
        let resultado = num1 - num2
        console.log(`Resultado ${resultado}`)
    }else if(op === "divisão"){
        if(num2 === 0){
            throw new Error('impossível dividir por zero, Einstein!?!')
        }else {
            let resultado = num1 / num2
            console.log(`Resultado ${resultado}`)
        }
    }else if(op === "multiplicação"){
        let resultado = num1 * num2
        console.log(`Resultado ${resultado}`)
    }

    
    
    if (isNaN(num1) || isNaN(num2)) {
        throw new Error('Digite números válidos.');
    }

   
} catch (erro) {
    // Capturar e lidar com exceções
    console.error('Ocorreu um erro:', erro.message);
} finally {
    // Código que será executado independentemente de ocorrer uma exceção ou não
    console.log('Operação concluída.');
}